src/lib/insights/client.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  Hits,
+  InsightsClient,
+  InsightsClientMethod,
+  InsightsClientPayload,
+  Renderer,
+  RenderOptions,
+  Unmounter,
+  WidgetFactory,
+} from '../../types';
+import find from 'lodash/find';
+
+export const inferPayload = (
+  method: InsightsClientMethod,
+  results: RenderOptions['results'],
+  hits: Hits<unknown>,
+  objectIDs: string[]
+): InsightsClientPayload => {
+  const { index, queryID } = results;
+  switch (method) {
+    case 'clickedObjectIDsAfterSearch': {
+      const positions = objectIDs.map(objectID => {
+        const hit = find(hits, hit => hit.objectID === objectID);
+        // NOTE: we might want to optimize this so as we browse though the
+        // hits once, which can get pretty big
```

Comment:
The maximum number of hits returned by the engine is 1000 I think. In the UI it's very rare that we display more than 100 hits at once so it's not a big concern IMO.